### ISSUE TITLE
Notebook 15 and the Pastas ARMA(1,1) noise model

### ISSUE THREAD

**Issue by tdmeij** [pastas/pastas#235]
Version 0.15.0 of Pastas provides an ARMA(1,1) noisemodel with more flexibility than the AR(1) model. Notebook 15 demonstrates this model and shows that noise parameters alpha and beta are estimated correctly for alpha = 0.8 and beta = 0.7

**user requests**
However, simulations on a broader range of alpha and beta values suggests that performance of the noise model is not optimal for all values of alpha and beta. Therefore, Notebook 15 could be extended with examples for areas with lesser performance to give users some insight in limitations of this model. 
In the current implementation, the ARMA(1,1) noisemodel in Pastas is restricted to beta values of zero and higher. Maybe, it would be better not to restrict the ARMA-model in Pastas, but to issue a user warning if calibrated parameter values go outside plausible values. This would make it possible to test the model on more data and find out under which circumstances calibration results in unlikely beta values.

**results of simulations**
The figures below show the results of simulations using the code in Notebook 15 for values of alpha between 0 en 0.999 and for beta between -0.9 and 0.9. Scripts used for simulations are available on GitHub: https://github.com/tdmeij/pastas-arma-simulations

Since the ARMA(1,1) model in Pastas is restricted to beta > 0 the simulations are expected to give unreliable results for values of true beta < 0. These deviations are the result of testing the model outside it's known limits.

**test 1: estimates of alpha and beta**
First, the heatmap below of the relative deviation of alpha_est and beta_est from their true values confirms that the estimated alpha and beta are within 10% of their true values, in most of the simulated range:

 ![image](https://user-images.githubusercontent.com/58006409/91350754-6c615080-e7e7-11ea-983d-63aed117725e.png)

The estimated alpha (left figure) deviates from it's true value for values of true alpha smaller than 0.3, regardless of the value of true beta. For values of true beta smaller than zero, the estimated alfa values are unreliable because of the restriction (beta>0) in the Pastas ARMA(1,1)-model. (Pastas returns 0 for the estimated beta in this region).

Deviations of the estimated beta from true beta (right figure) are negligible in most of the the region with true beta>0. For combinations of  true alpha's smaller than approximately 0.7 and true beta's smaller than approximately 0.3, deviations of the estimated beta become larger than 10%. 

**test 2: estimates of the gain**
The next figure below shows the estimated gain (A) of the Pastas model (the true value for all models is A=800). The true gain is estimated correctly for most simulations, with exception of the models with a true noise parameter alpha > 0.8. In these cases the estimated gain A of the deterministic model seems to be negatively affected by the noise model.
 
![image](https://user-images.githubusercontent.com/58006409/91350703-594e8080-e7e7-11ea-8dc7-2c387978d202.png)

**test 3: normality and autocorrelation in innovations**
Innovations of a calibrated noise-model should preferably be normally distributed and free from autocorrelation. Pastas calculates some test statistics with ps.stats.diagnostics(). The heatmaps below show the results of tests for normality and autocorrelation. These figures suggest that normality and absence of autocorrelation are not guaranteed over the entire simulated range of alpha and beta.

![image](https://user-images.githubusercontent.com/58006409/91350638-420f9300-e7e7-11ea-888e-33d867cc2bf9.png)

![image](https://user-images.githubusercontent.com/58006409/91350558-24dac480-e7e7-11ea-9743-a77ed844aa4c.png)
 

**Use of the ARMA(1,1) noise model**
The simulation results suggest that the ARMA(1,1) noise model as currently implemented in Pastas can be used in the region with true alfa's smaller than 0.8 and true beta's larger than 0.3  Although estimates for beta are given in the region for true beta's between 0 and 0.3, models in this region should be treated with care since reliable estimates of alpha and A in this region with true beta<0.3 can only be expected for true alpha's between 0.7 and 0.8. Given these results, it would be interesting to know what ranges of true alpha's and true beta's can be expected in real world applications. For the simulated data, innovations are normally distributed and free from autocorrelation over a limited range of alpha and beta only.

**Comment by raoulcollenteur** [comment]
Hi Thomas! Thanks for the analysis and the issue report, I'm gonna read through it carefully tomorrow. It looks really interesting 👍

Just one quick thing that came to my mind earlier today. I think one important factor here is that with such high values for alpha and beta the standard deviation of the residual series is relatively large compared to the standard deviation of the groundwater level fluctuations. Such that even the perfect model will have a low EVP (e.g., 50%). In other words, higher values of alpha will lead to a different distributions of the standard deviations of the residuals and the groundwater levels, or in other words lower EVP (assuming we are using the same value for noise_perc in gwsym.py). Maybe this also influences when the parameters are correctly estimated or not? At least in- or decreasing noise_perc will change the boundaries for which beta and alpha are correctly estimated in the synthetic example so it might be hard to provide users with a single value.

I have an easy fix for the negative beta values, but need to look into it a little more before implementing it, not sure yet what negative beta values would "mean".

To be continued :)
Raoul

**Comment by tdmeij** [comment]
Hi Raoel

Thanks for your fast reply. For the moment I'am working on more practical
applications with Pastas, so I'am not really waiting for a fast reply to
the issue report. It will be interesting to see your changes in the model
though, meanwhile I am working on models of groundwater series of provincie
Overijssel for practical use. Thanks for developing Pastas!

greetings, Thomas




Op wo 26 aug. 2020 om 23:02 schreef Raoul Collenteur <
notifications@github.com>:

**Comment by raoulcollenteur** [comment]
Hi Thomas,

I finally got to look at the scripts and the problems you indicated with the ARMA(1,1) model and finding parameter values. Some decisions made during model optimisation (deviations from the default settings) caused the problem that the original parameters were not found back. The analyses can be found in the forked repo: https://github.com/pastas/pastas-arma-simulations

**Changes made**
- Use of warmup, remove tmin and tmax. The use of a warmup period is important, but was set to 0 in the original scripts. For higher values of alpha and beta, not providing the entire time series (e.g., using tmax and tmin) caused lower performance.
- I changed recharge_a to 50, a more realistic value. The chosen recharge_a was quite high, especially without providing better initial parameters (for example by solving without a noisemodel first). It actually still works well with 200, but I think with such large deviations from the initial parameters you would/should provide better initial estimates in real life cases. 

**Findings**
- only at very high values of alpha and beta the estimates of A differ more, but the true A is still within the confidence intervals of the estimated parameter. That means that with repeated simulations the mean may still be close to the true A. Repeating simulations would actually be interesting (now a fixed random seed is chosen), but adds another dimensionality and complexity to the analysis.
- The added noise for these models with large alpha and beta is disproportionally large, probably hardening the search for the true parameters. E.g. variance of the noise is twice the variance of the groundwater levels.
- at alpha = -beta the two terms cancel out and both terms can also be zero because you end up with the random noise you put in. Although this diagonal shows large differences there is nothing wrong with the models on this diagonal. See also e.g., http://www.maths.qmul.ac.uk/~bb/TimeSeries/TS_Chapter4_6.pdf

<img src="https://user-images.githubusercontent.com/12051167/94693665-48d47d00-0334-11eb-9ffc-c1198072119f.png" alt="drawing" width="400"/>

<img src="https://user-images.githubusercontent.com/12051167/94695461-3a876080-0336-11eb-9fd6-e6e3f24f0efc.png" alt="drawing" width="700"/>

**Other changes**
- I modified the code such that it can handle negative Beta's (ba439fb), thanks for the suggestion! Works well as shown in the figure above.

**Most important conclusion, I think for most (realistic) ranges of beta and alpha the model parameters are reliably estimated.**

Thanks again for testing the new model on synthetic time series! Let me know if this solves the issue, if so I'll close the Issue later this week :)
